### PR TITLE
Add inline prediction option mirroring the capitalization option

### PR DESCRIPTION
**Pull request by 347Online**

### CODE REVIEW

**Comment by emilazy** [comment]
Thank you for the contribution! Looks like the tests are failing, unfortunately.

**Comment by 347Online** [comment]
This is true. I haven't figured out why yet. I've been meaning to take another look but then I got busy 🙃

File: tests/fixtures/system-defaults-write/activate-user.txt
Comment (emilazy) [inline tests/fixtures/system-defaults-write/activate-user.txt:1]: Just a simple ordering issue:

```diff
diff --git a/tests/fixtures/system-defaults-write/activate-user.txt b/tests/fixtures/system-defaults-write/activate-user.txt
index c22b0a3ff6...d324a8d909 100644
--- a/tests/fixtures/system-defaults-write/activate-user.txt
+++ b/tests/fixtures/system-defaults-write/activate-user.txt
@@ -68,16 +68,16 @@
 <plist version="1.0">
 <false/>
 </plist>'
+defaults write -g 'NSAutomaticDashSubstitutionEnabled' $'<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<false/>
+</plist>'
 defaults write -g 'NSAutomaticInlinePredictionEnabled' $'<?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
 <false/>
 </plist>'
-defaults write -g 'NSAutomaticDashSubstitutionEnabled' $'<?xml version="1.0" encoding="UTF-8"?>
-<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-<plist version="1.0">
-<false/>
-</plist>'
 defaults write -g 'NSAutomaticPeriodSubstitutionEnabled' $'<?xml version="1.0" encoding="UTF-8"?>
 <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
 <plist version="1.0">
```

Happy to merge if you amend your commit with this.

**Review by emilazy — CHANGES_REQUESTED** [review]

File: tests/fixtures/system-defaults-write/activate-user.txt
Comment (347Online) [reply tests/fixtures/system-defaults-write/activate-user.txt:1]: Ooh okay, thank you!

**Review by emilazy — CHANGES_REQUESTED** [review]
Looks like all the tests are passing now!

Sorry to nitpick, but we like to maintain a linear history (no merges from master into pull requests) and keep fix‐ups in the same commit (to make `git bisect` work better and scanning the commit log cleaner). Could you rebase on the upstream commit and squash your two commits together so I can merge this?

```console
$ git rebase b2ee0b3c03b06bd70e4280c65ab8803c3456bb0f
$ git rebase -i HEAD~2
(change "pick" to "f"(ixup) on the fix commit)
```

**Comment by 347Online** [comment]
Ah, alright, no problem. I'll do that when I get back to my computer.

**Comment by 347Online** [comment]
I _think_ that should be done, I really appreciate the assist!

Please let me know if I need to make any additional changes, thank you! :)

**Comment by emilazy** [comment]
Looks like you accidentally moved the new upstream commits after yours – if you run `git rebase -i 29d16ceb7c13c2741d10f77230caca6163cfa904^` you should be able to just remove all the lines other than your commit and resolve the conflict. Git’s UI is pretty terrible, sadly…

**Comment by 347Online** [comment]
Sorry for the delayed response - I tried your latest suggestion to no avail, I assume I must not be understanding something? 🤔

I'll keep tinkering

**Comment by emilazy** [comment]
Git’s UI is really awful, it’s not your fault. (I’ll take this opportunity to plug [Jujutsu](https://github.com/martinvonz/jj), which makes all of this a lot simpler and easier to recover from; I can’t imagine using Git’s UI directly any more.)

In general when working on Nix ecosystem repositories you almost always want to `git pull --rebase` and not introduce any merge commits, and liberally use `git commit --amend` and `git rebase --interactive` to squash fixes into the original commits. The easiest way to get out of this mess is probably as follows:

```
# Get the latest revision from LnL7/nix-darwin into FETCH_HEAD
# Replace “upstream” with whatever your https://github.com/LnL7/nix-darwin.git remote is
# (`git remote list` to find out, `git remote add` if you don’t have one)
$ git fetch upstream master

# Put your current branch at FETCH_HEAD without changing your working tree
$ git reset FETCH_HEAD

# Record every change between upstream and your current files as a single commit
$ git commit -a

# Check the commit turned out right; if something looks wrong, adjust your files
# and `git commit --amend -a` until it looks good
$ git show HEAD

# Update your pull request
$ git push --force-with-lease
```

**Comment by 347Online** [comment]
Jujutsu sounds VERY cool, I'll have to give it a look

I'm traveling for work this week, but I'll definitely give your suggestions a try when I can.

I really am sorry about all this, I can't tell you how much I appreciate all your help, thank you so much <3

**Review by emilazy — APPROVED** [review]
Looks good, thank you!

**Pull request merged**